images/recycler/scripts/recycler.sh

```diff
@@ -22,8 +21,10 @@ if [[ ! -d ${dir} ]]; then
     exit 1
 fi
 
-# shred all files
-find ${dir} -type f -exec shred -fuvz {} \;
+find ${dir} -type f -exec stat -c "%u %g %n"  {} \; |xargs -n 3 bash -c 'sudo -u "#$1" shred  $3' argv0
```

Comment:
In general, I'm concerned that this takes user-controlled data (filenames) and makes them input to a bash script running in a pod with hostmount access, likely as uid 0.